### ISSUE TITLE
BUG: Problems when using tqdm

### ISSUE THREAD

**Issue by schlichtanders** [computationalmodelling/nbval#56]
Dear nbval-Team,

unfortunately nbval is not working with ``tqdm``. (this is the minimal working example for #55)

While a cell with ``output_type="execute_result"`` seem to always work, like e.g.
```
s = "foo"
s
```
the success of a cell with ``output_type="stream"`` depends on whether ``tqdm`` was imported or not. For instance
```
import tqdm
print("goo")
```
will throw the following error when tested with ``py.test --nbval  path/to/file.ipynb``
```
py.test --nbval .\test_nbval.ipynb
============================= test session starts =============================
platform win32 -- Python 3.6.0, pytest-3.0.5, py-1.4.32, pluggy-0.4.0
rootdir: D:\Projects\telefonica-callcenter-online, inifile:
plugins: nbval-0.5
collected 3 items 

test_nbval.ipynb ..F

================================== FAILURES ===================================
___________________________________ cell 2 ____________________________________
Notebook cell execution failed
Cell 2: Cell outputs differ

Input:
import tqdm
print("goo")

Traceback: mismatch 'output_type'
<<<<<<<<<<<< Reference output from ipynb file:
stream
============ disagrees with newly computed (test) output:  
streamstream
>>>>>>>>>>>>
===================== 1 failed, 2 passed in 2.13 seconds ======================
```
This behaviour does not depend at all on where ``tqdm`` is imported but only on whether ``tqdm`` gets imported. Without, everything runs through as expected.

I created a small gist with a respective example notebook, you can find it here https://gist.github.com/anonymous/15a17512fb72fee2c8a8abbe46316eff

Any help is highly appreciated, as I love this pytest extension and would really like to use it more.

**Comment by vidartf** [comment]
Seems strongly related (duplicate?) to #54.

**Comment by schlichtanders** [comment]
that is true, sorry for not reading over the issues

Reading over #54 it seems like tqdm changes printing to refer to multiple streams in the background. However considering the output and the .ipynb file nothing changes when adding the tqdm import.

Isn't it possible to change nbval to just look at the actual notebook output?

**Comment by vidartf** [comment]
This is definitely something that should be improved upon, and I hope to have a PR for this tomorrow.

**Comment by schlichtanders** [comment]
;-) unfortunately I don't have enough time at the moment to fix it myself

**Comment by vidartf** [comment]
Fix suggested in #57.

**Comment by takluyver** [comment]
Sorry, I thought #57 was the one we already merged.

**Comment by vidartf** [comment]
Fixed in #57.
